### PR TITLE
Anime Shitai Presence

### PR DESCRIPTION
**Pull request by priim0**
home

### CODE REVIEW

File: Anime Shitai/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "description": {
+    "en": "Anime Shitai is a German Anime Source where you can watch Animes here with german subtitle and german dubbed. We have over 1000 Animes that wait for you to watch!",
+    "de": "Anime Shitai ist eine Anime Stream Source wo ihr über 1000 kostenlose Anime Serien anschauen könnt, Jede woche gibt auch noch die neuste Anime Folgen frisch aus Japan mit deutschem Untertitel nur für euch und das völlig Kostenlos auf Full HD / 1080p ",
+    "fr": "Anime Shitai est un portail d'anime allemand où vous pouvez regarder des animes en allemand ou avec des sous-titres allemands. Nous avons plus de 1000 animes qui vous attendent!"
+  },
+	"url": [
+    "www.anime-shitai.com",
+    "anime-shitai.com"
+  ],
+  "version": "1.3.8",
```
Comment (Bas950) [inline Anime Shitai/dist/metadata.json:17]: version 1.0.0 on release

File: Anime Shitai/dist/presence.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    clientId: "814910404429217852"
+}), strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused"
+});
+function getTimestamps(videoTime, videoDuration) {
+    var startTime = Date.now();
+    var endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+    return [Math.floor(startTime / 1000), endTime];
+}
```
Comment (Bas950) [inline Anime Shitai/dist/presence.js:11]: use presence.getTimestamps()

**Review by Bas950 — CHANGES_REQUESTED** [review]
no var, dont upload js

File: Anime Shitai/dist/metadata.json (hunk 1 above)
Comment (priim0) [reply Anime Shitai/dist/metadata.json]: is now changed

File: Anime Shitai/dist/presence.js (hunk 2 above)
Comment (priim0) [reply Anime Shitai/dist/presence.js]: sure, i changed it, now must be okay

**Comment by priim0** [comment]
and now?

**Review by Bas950 — CHANGES_REQUESTED** [review]
no var, dont upload js, fix codacy issues

File: Anime Shitai/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "en": "Anime Shitai is a German Anime Source where you can watch Animes here with german subtitle and german dubbed. We have over 1000 Animes that wait for you to watch!",
+    "de": "Anime Shitai ist eine Anime Stream Source wo ihr über 1000 kostenlose Anime Serien anschauen könnt, Jede woche gibt auch noch die neuste Anime Folgen frisch aus Japan mit deutschem Untertitel nur für euch und das völlig Kostenlos auf Full HD / 1080p ",
+    "fr": "Anime Shitai est un portail d'anime allemand où vous pouvez regarder des animes en allemand ou avec des sous-titres allemands. Nous avons plus de 1000 animes qui vous attendent!"
+  },
+	"url": [
+    "www.anime-shitai.com",
+    "anime-shitai.com"
+  ],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/8PWKaRe.jpg",
```
Comment (StrikerFRFX) [inline Anime Shitai/dist/metadata.json:18]: Is this the best logo you could find? It looks like one of those Fake PNGs you find on Google

File: Anime Shitai/iframe.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
+var iframe = new iFrame();
```
Comment (StrikerFRFX) [inline Anime Shitai/iframe.ts:1]: No var, const

File: Anime Shitai/iframe.ts [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+iframe.on("UpdateData", async () => {
+  var video: HTMLVideoElement;
```
Comment (StrikerFRFX) [inline Anime Shitai/iframe.ts:3]: No var, let

File: Anime Shitai/presence.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
+var presence = new Presence({
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:1]: No var, use const

File: Anime Shitai/presence.ts [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    clientId: "814910404429217852" // CLIENT ID FOR YOUR PRESENCE
+  }), 
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused"
+  });
+
+/**
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:9]: Remove this entire function and its comments, use `presence.getTimestamps`

File: Anime Shitai/presence.ts [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  var startTime = Date.now();
+  var endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
+
+var browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:23]: No var and merge all these variables

File: Anime Shitai/presence.ts [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+    iFrameVideo = data.iframe_video.iFrameVideo;
+    currentTime = data.iframe_video.currTime;
+    duration = data.iframe_video.dur;
+    paused = data.iframe_video.paused;
+  }
+});
+
+ 
+presence.on("UpdateData", async () => {
+	  var timestamps = getTimestamps(Math.floor(currentTime), Math.floor(duration)),
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:49]: No var const or let

File: Anime Shitai/presence.ts [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+    presence.setActivity(presenceData);
+  } else {
+    presenceData.details = "Chillt irgendwo auf der Seite.";
+    delete presenceData.state;
+    delete presenceData.endTimestamp;
+    presenceData.startTimestamp = browsingStamp;
+    delete presenceData.smallImageText;
+    presence.setActivity(presenceData);
+  }
+
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:165]: Remove newlines

File: Anime Shitai/presence.ts [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+});
+
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:167]: Newline

File: Anime Shitai/presence.ts [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+    delete presenceData.state;
+    delete presenceData.endTimestamp;
+    presenceData.startTimestamp = browsingStamp;
+    delete presenceData.smallImageText;
+    presence.setActivity(presenceData);
+  }
+
+});
+ 
+
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:168]: Newline

File: Anime Shitai/presence.ts [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+
```
Comment (StrikerFRFX) [inline Anime Shitai/presence.ts:169]: Newline

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
Please see my comments.

File: Anime Shitai/dist/metadata.json (hunk 3 above)
Comment (priim0) [reply Anime Shitai/dist/metadata.json:18]: Its not a Fake PNG, its the logo from our website since years.

Comment (StrikerFRFX) [reply Anime Shitai/dist/metadata.json:18]: Hmm okay, could probably get a better quality image for the sake of consistency if a better quality one exists.

**Comment by Bas950** [comment]
Closed due to inactivity.

**Pull request closed without merging**